handles.go

```diff
@@ -0,0 +1,198 @@
+// Copyright 2021 The Go Authors. All rights reserved.
```

Comment:
Could we make all the symbols in this file unexported (so we'll not have to maintain backward compatibility for them for the users of the library)?

Reply:
good call!